Source/embeddings/Builder/embeddingDecorator.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { Guid } from '@dolittle/rudiments';
+import { ScopeId } from '@dolittle/sdk.events';
+import { ProjectionDecoratedType, ProjectionDecoratedTypes, ProjectionId } from '@dolittle/sdk.projections';
+
+import { EmbeddingId } from '..';
+
+import { EmbeddingDecoratedType } from './EmbeddingDecoratedType';
+import { EmbeddingDecoratedTypes } from './EmbeddingDecoratedTypes';
+
+/**
+ * Decorator to mark a class as an Embedding.
+ * @param {EmbeddingId | Guid | string} id The id to associate with this Embedding
+ */
+export function embedding(id: EmbeddingId | Guid | string) {
+    return function (target: any) {
+        const embeddingId = EmbeddingId.from(id);
+        EmbeddingDecoratedTypes.register(new EmbeddingDecoratedType(
+            embeddingId,
+            target));
+        ProjectionDecoratedTypes.register(new ProjectionDecoratedType(
+            ProjectionId.from(embeddingId.value),
+            ScopeId.default,
+            target));
```

Comment:
Honestly I can't remember for sure at this point. I think my idea was so that you can't register a class as both an embedding and as a projection. But now that I look at this this might actually break when built.

Reply:
I guess we can try it out. If it works then I have nothing against it